test/tests/unit/measurement/store.test.ts

```diff
@@ -223,103 +331,57 @@ describe('measurement store', () => {
 
-	it('should store measurement probes in the same order as in arguments', async () => {
+	it('should initialize measurement object with the proper default in case of offline probes', async () => {
 		const store = getMeasurementStore();
 		await store.createMeasurement(
 			{
 				type: 'ping',
 				measurementOptions: { packets: 3 },
 				target: 'jsdelivr.com',
 				locations: [],
-				limit: 4,
+				limit: 1,
 				inProgressUpdates: false,
 			},
-			[ getProbe('z'), getProbe('10'), getProbe('x'), getProbe('0') ],
+			new Map(),
+			[ getOfflineProbe('id', '1.1.1.1') ],
 		);
 
-		expect(redisMock.hSet.callCount).to.equal(1);
-		expect(redisMock.hSet.args[0]).to.deep.equal([ 'gp:in-progress', 'measurementid', 1678000000000 ]);
-		expect(redisMock.set.callCount).to.equal(1);
-		expect(redisMock.set.args[0]).to.deep.equal([ 'gp:measurement:measurementid:probes_awaiting', 4, { EX: 35 }]);
-		expect(redisMock.json.set.callCount).to.equal(1);
-
-		expect(redisMock.json.set.args[0]).to.deep.equal([ 'gp:measurement:measurementid', '$', {
-			id: 'measurementid',
-			type: 'ping',
-			status: 'in-progress',
-			createdAt: '2023-03-05T07:06:40.000Z',
-			updatedAt: '2023-03-05T07:06:40.000Z',
-			target: 'jsdelivr.com',
-			limit: 4,
-			probesCount: 4,
-			results: [{
-				probe: {
-					continent: 'continent',
-					region: 'region',
-					country: 'country',
-					state: 'state',
-					city: 'city',
-					asn: 'asn',
-					longitude: 'longitude',
-					latitude: 'latitude',
-					network: 'z',
-					tags: [],
-					resolvers: [],
-				},
-				result: { status: 'in-progress', rawOutput: '' },
-			},
-			{
-				probe: {
-					continent: 'continent',
-					region: 'region',
-					country: 'country',
-					state: 'state',
-					city: 'city',
-					asn: 'asn',
-					longitude: 'longitude',
-					latitude: 'latitude',
-					network: '10',
-					tags: [],
-					resolvers: [],
-				},
-				result: { status: 'in-progress', rawOutput: '' },
-			},
+		expect(redisMock.json.set.firstCall.args).to.deep.equal([
+			'gp:measurement:measurementid',
+			'$',
 			{
-				probe: {
-					continent: 'continent',
-					region: 'region',
-					country: 'country',
-					state: 'state',
-					city: 'city',
-					asn: 'asn',
-					longitude: 'longitude',
-					latitude: 'latitude',
-					network: 'x',
-					tags: [],
-					resolvers: [],
-				},
-				result: { status: 'in-progress', rawOutput: '' },
+				id: 'measurementid',
+				type: 'ping',
+				status: 'in-progress',
+				createdAt: '2023-03-05T07:06:40.000Z',
+				updatedAt: '2023-03-05T07:06:40.000Z',
+				target: 'jsdelivr.com',
+				probesCount: 1,
+				results: [
+					{
+						probe: {
+							continent: 'continent',
+							region: 'region',
+							country: 'country',
+							state: 'state',
+							city: 'city',
+							asn: 'asn',
+							longitude: 'longitude',
+							latitude: 'latitude',
+							network: 'id',
+							tags: [],
+							resolvers: [],
+						},
+						result: {
+							status: 'failed',
```

Comment:
Do you think it isn't worth creating its own status value in the end?

Reply:
Initially I thought yes. But maybe for some UI with a list of probes constantly pinging it will be nice to mark either probe is offline or measurement failed. In that case make a condition based on `rawOutput` isn't nice, so introducing `status: offline` makes sense.